### PR TITLE
Make sure `selectedItem` is cleared

### PR DESCRIPTION
**Pull request by blobor**
Add check for an empty string (to make sure we didn't add empty pill), also make sure we clean it after selection.

### CODE REVIEW

File: modules/TokenizedSelect.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ import classnames from 'classnames'
 import TokenWrap from './TokenWrap'
 import Item from './Item'
 
+const isNotEmptyString = str => str != null && str.length !== 0
```
Comment (beardedtim) [inline modules/TokenizedSelect.js]: can't you just do `str => str` and it work the same since an empty string returns falsey?

Comment (jgbenito7) [reply modules/TokenizedSelect.js]: Who is this Tim person??

Comment (blobor) [reply modules/TokenizedSelect.js]: In this case name of the function is confusing, cause if function starts with `is` letters we should expect Boolean value.

And yes, how is this guy 😆? Do we know him?

Comment (beardedtim) [reply modules/TokenizedSelect.js]: I'm the outcast here but I like either the consumer deciding what to do with truthy/falsey values or the double `!!` to transform it into a Boolean:

```
str => !!str
```

File: modules/TokenizedSelect.js [hunk 2]
```diff
@@ -7,6 +7,8 @@ import classnames from 'classnames'
 import TokenWrap from './TokenWrap'
 import Item from './Item'
 
+const isNotEmptyString = str => !!str
```
Comment (jgbenito7) [inline modules/TokenizedSelect.js]: an empty string will already return false

File: modules/TokenizedSelect.js [hunk 3]
```diff
@@ -43,7 +45,9 @@ class TokenizedSelect extends Component {
   }
 
   handleChange = selectedItem => {
-    this.addSelectedItem(selectedItem)
+    if (isNotEmptyString(selectedItem)) {
```
Comment (jgbenito7) [inline modules/TokenizedSelect.js]: this can just be `if (selectedItem)`

Comment (blobor) [reply modules/TokenizedSelect.js]: How about `0` value?

File: modules/TokenizedSelect.js (hunk 2 above)
Comment (blobor) [reply modules/TokenizedSelect.js]: Yeah, you are right. I will revert my changes to previous.

File: modules/TokenizedSelect.js (hunk 1 above)
Comment (blobor) [reply modules/TokenizedSelect.js]: As mentioned by @jgbenito7 https://github.com/kofile/react-tokenized-select/pull/4#discussion_r159965184, i'm reverting my changes

**Pull request merged**